[Docs] Expand zone-aware replication page to cover single physical availability zone deployments. (#4631)
grafana/mimir: docs/sources/mimir/configure/configure-zone-aware-replication.md

> Ensure that you configure deployment tooling so that it is also zone-aware.
> The deployment tooling is responsible for executing rolling updates.
> Rolling updates should only update replicas in a single zone at any given time.

> > You can utilize the [Kubernetes rollout operator](#kubernetes-operator-for-simplifying-rollouts-of-zone-aware-components) to assist with this.

Reviewer: Capitalize here to make a distinction between a Kubernetes Operator and a human operator.
```suggestion
> You can utilize the [Kubernetes rollout Operator](#kubernetes-operator-for-simplifying-rollouts-of-zone-aware-components) to assist with this.
```

The [Kubernetes Rollout Operator](https://github.com/grafana/rollout-operator) is a Kubernetes operator that makes it easier for you to manage multi-availability-zone rollouts. Consider using the Kubernetes Rollout Operator when you run Grafana Mimir on Kubernetes with zone awareness enabled.

> Note that even in a scenario where you have deployed Grafana Mimir to a single physical availability zone, the Rollout Operator is immensely useful.

Reviewer: ^ Could you please capitalize Operator in line 90 to follow the standard we've set for capitalizing K8 Operators to distinguish from human operators.

> This is because it automates and accelerates the process of rollouts by:

Reviewer: ```suggestion
This is because the Operator automates and accelerates the process of rollouts by:
```

> - Allowing an entire zone to be updated at a time

Reviewer: ```suggestion
- Allowing an entire zone to be updated at one time
```

- Prioritizing rollout of unhealthy zones over healthy zones to ensure stability
- Allowing the administrator to determine how aggressive rollouts should be

> This functionality is particularly useful in larger deployments where rollouts would otherwise take a very long time and likely require manual intervention if issues were to occur.

Reviewer: ```suggestion
The automated functionality provided by an Operator is particularly useful in larger deployments where rollouts would otherwise take a very long time and likely require manual intervention if issues were to occur.
```
